File: 800/Java/P14.java
import java.util.*;

public class P14 {
    static void solve(Scanner scanner) {
        long a = scanner.nextLong();
        long b = scanner.nextLong();
        long c = scanner.nextLong();

        if(a > b)   System.out.println("First");
        else if(b > a)   System.out.println("Second");
        else {
            if(c % 2 == 1)   System.out.println("First");
            else    System.out.println("Second");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while(t-- > 0)  solve(scanner);
        scanner.close();
    }
}

File: 800/C++/p14.cpp
#include<bits/stdc++.h>
using namespace std;

void solve() {
    long a, b, c;
    cin>>a>>b>>c;

    if(a > b)   cout<<"First"<<endl;
    else if(b > a)  cout<<"Second"<<endl;
    else {
        if(c % 2 == 1)  cout<<"First"<<endl;
        else    cout<<"Second"<<endl;
    }
}

int main() {
    int t;
    cin>>t;
    while(t--)  solve();
    return 0;
}